builder/qemu/step_create_disk.go

```diff
@@ -12,28 +12,36 @@ import (
 
 // This step creates the virtual disk that will be used as the
 // hard drive for the virtual machine.
-type stepCreateDisk struct{}
+type stepCreateDisk struct {
+	AdditionalDiskSize []string
+	DiskImage          bool
+	DiskSize           string
+	Format             string
+	OutputDir          string
+	UseBackingFile     bool
+	VMName             string
+	QemuImgArgs        QemuImgArgs
```

Comment:
This is much better! It makes a lot of sense to pass what's used in the step here instead of just using the config from the state 👏